packages/portal/src/utils/usePortal.ts

```diff
@@ -0,0 +1,90 @@
+import { useRef, useEffect, MutableRefObject } from 'react';
+
+import { Maybe } from '../interfaces';
+
+
+/**
+ * Creates DOM element to be used as React root.
+ * @returns {HTMLElement}
+ */
+export const createRootElement = (id: string, zIndex: number | string): HTMLElement => {
+  const rootContainer = document.createElement('div');
+  rootContainer.setAttribute('id', id);
+  rootContainer.setAttribute('style', `z-index: ${zIndex};`);
+  return rootContainer;
+};
+
+/**
+ * Appends element as last child of body.
+ * @param {HTMLElement} rootElem
+ */
+export const addRootElement = (rootElem: HTMLElement) => {
+  if (document.body.lastElementChild) {
+    document.body.insertBefore(
+      rootElem,
+      document.body.lastElementChild.nextElementSibling,
+    );
+  }
+};
+
+/**
+ * Hook to create a React Portal.
+ * Automatically handles creating and tearing-down the root elements (no SRR
+ * makes this trivial), so there is no need to ensure the parent target already
+ * exists.
+ * @example
+ * const target = usePortal(id, [id]);
+ * return createPortal(children, target);
+ * @param {String} id The id of the target container, e.g 'modal' or 'spotlight'
+ * @returns {HTMLElement} The DOM node to use as the Portal target.
+ */
+export const usePortal = (id: string, zIndex: number | string = 0) => {
```

Comment:
Why usePortal accept zIndex (it is not his logic)?
You could add zIndex to component, that will be used inside portal.

Reply:
Because it would be needed manage position on layout of another DOM element which would be root element for rendering